certora/specs/Reverts.spec

```diff
@@ -0,0 +1,253 @@
+    require collateralQuoted > 0;
+
+    mathint borrowed = require_uint256(summaryToAssetsUp(pBorrowShares, mTotalBorrowAssets, mTotalBorrowShares));
+    mathint ltv = require_uint256(wDivUp(require_uint256(borrowed), require_uint256(collateralQuoted)));
+
+    preLiquidate@withrevert(e, borrower, seizedAssets, 0, data);
+
+    // Ensure the price is unchanged in the preLiquidate call.
+    require !priceChanged;
+
+    assert  ltv > currentContract.LLTV => lastReverted;
+}
+
+rule excessivePreliquidationReverts(env e,address borrower, uint256 seizedAssets, bytes data) {
+    uint256 mTotalBorrowAssets;
+    uint256 mTotalBorrowShares;
+    uint256 mLastUpdate;
+
+    uint256 pBorrowShares;
+    uint256 pCollateral;
+
+    (_, _, mTotalBorrowAssets,mTotalBorrowShares,mLastUpdate, _) = MORPHO.market(currentContract.ID);
+
+    // Ensure that no interest is accumulated.
+    require mLastUpdate == e.block.timestamp;
+
+    uint256 collateralPrice = mockPrice();
+
+    (_, pBorrowShares, pCollateral) = MORPHO.position(currentContract.ID, borrower);
+
+    mathint collateralQuoted = require_uint256(summaryMulDivDown(pCollateral, collateralPrice, ORACLE_SCALE()));
+
+    // Safe require because the implementation would revert, see rule zeroCollateralQuotedReverts.
+    require collateralQuoted > 0;
+
+    requireInvariant preLltvConsistent();
+    requireInvariant preLCFConsistent();
+    requireInvariant preLIFConsistent();
+
+    mathint borrowed = require_uint256(summaryToAssetsUp(pBorrowShares, mTotalBorrowAssets, mTotalBorrowShares));
+    mathint ltv = require_uint256(wDivUp(require_uint256(borrowed), require_uint256(collateralQuoted)));
+
+
+    mathint preLIF = computeLinearCombination(ltv,
+                                              currentContract.LLTV,
+                                              currentContract.PRE_LLTV,
+                                              currentContract.PRE_LIF_1,
+                                              currentContract.PRE_LIF_2) ;
+
+    // Safe require as implementation would revert with InconsistentInput.
+    require seizedAssets > 0;
+
+    mathint seizedAssetsQuoted = require_uint256(summaryMulDivUp(seizedAssets, collateralPrice, ORACLE_SCALE()));
+
+    mathint repaidShares = summaryToSharesUp(wDivUp(require_uint256(seizedAssetsQuoted), require_uint256(preLIF)),
+                               mTotalBorrowAssets,
+                               mTotalBorrowShares);
+
+    mathint closeFactor = computeLinearCombination(ltv,
+                                                   currentContract.LLTV,
+                                                   currentContract.PRE_LLTV,
+                                                   currentContract.PRE_LCF_1,
+                                                   currentContract.PRE_LCF_2) ;
+
+    mathint repayableShares = wMulDown(pBorrowShares, require_uint256(closeFactor));
+
+    preLiquidate@withrevert(e, borrower, seizedAssets, 0, data);
```

Comment:
could we do the same proof for repaidShares as input? Should be even simpler no?